packages/buidler-core/src/internal/core/providers/http.ts

```diff
@@ -125,7 +134,7 @@ export class HttpProvider extends EventEmitter {
   }
 }
 
-export function isValidJsonResponse(payload: any) {
+export function isValidJsonRequest(payload: any): boolean {
```

Comment:
We should probably move `isValidJsonRequest` and `isValidJsonResponse` to something like `packages/buidler-core/src/internal/util/jsonrpc.ts`

Reply:
Also `JsonRpcRequest` and `JsonRpcResponse`. 

Importing this things from `packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts` is odd.

Reply:
Moved to `packages/buidler-core/src/internal/buidler-evm/jsonrpc/handler.ts`